src/main/java/com/google/sps/data/DatastoreClientInterface.java

```diff
@@ -0,0 +1,54 @@
+package com.google.sps.data;
+
+import java.util.Date;
+import com.google.appengine.api.datastore.Entity;
+import java.util.Optional;
+import java.util.List;
+
+/** Interface for the DatastoreClient class. */
+public interface DatastoreClientInterface {
+  /**
+   * Adds an AttendeeInterface object to the Datastore. If object already
+   * exist, the old one is replaced.  
+   */
+  void insertOrUpdateAttendee(AttendeeInterface attendee);
+
+  /**
+   * Adds an InstanceInterface object to the Datastore. If object already
+   * exist, the old one is replaced. 
+   */
+  void insertOrUpdateInstance(InstanceInterface instance);
+
+  /** 
+   * Adds an SessionInterface object to the Datastore. If object already 
+   * exist, the old one is replaced. 
+   */
+  void insertOrUpdateSession(SessionInterface session);
+
+   /** 
+    * Returns the Session object associated with the given sessionId. If 
+    * return Optional is empty, then no object exists with given parameter.
+    */
+  Optional<Session> getSession(String sessionId);
+
+  /**
+   * Returns the Instance object associated with the given instanceName. If 
+   * return Optional is empty, then no object exists with given parameter.
+   */
+  Optional<Instance> getInstance(String instanceName);
```

Comment:
Optional\<InstanceInterface\>

Make the changes throughout. You should be returning the interfaces, not the concrete classes